test/linear_combination.jl

```diff
@@ -9,8 +9,8 @@
         @testset "all inds" begin
             x = [1, 2]
             lc = LinearCombination([1, -1])
-            @test FeatureTransforms.apply(x, lc) == fill(-1)
-            @test lc(x) == fill(-1)
+            @test FeatureTransforms.apply(x, lc; dims=1) == fill(-1)
```

Comment:
LinearCombination no longer has a default `dim`, which is actually breaking come to think of it. Unless we add a deprecation somewhere.

Reply:
I've added a commit reversing the changes to the linear combination tests and instead added a deprecation warning for LinearCombination which we can remove in the next breaking release.